erpnext/controllers/queries.py

```diff
@@ -550,50 +550,32 @@ def item_manufacturer_query(doctype, txt, searchfield, start, page_len, filters)
 @frappe.whitelist()
 def get_purchase_receipts(doctype, txt, searchfield, start, page_len, filters):
 	fields = get_fields("Purchase Receipt", ["name"])
+	item_filters = [
+		['Purchase Receipt', 'docstatus', '=', '1'],
+		['Purchase Receipt Item','name', 'like', '%' + txt + '%']
+	]
 
-	query = """
-		select %(fields)s
-		from `tabPurchase Receipt` pr, `tabPurchase Receipt Item` pritem
-		where pr.docstatus = 1 and pritem.parent = pr.name
-		and pr.name like %(txt)s"""
-
-	flt = {
-		"fields": ", ".join(['pr.{0}'.format(f) for f in fields]),
-		"txt": frappe.db.escape('%{0}%'.format(txt))
-	}
-
-	if filters and filters.get('item_code'):
-		query += " and pritem.item_code = %(item_code)s"
-		flt.append({
-			"item_code": frappe.db.escape(filters.get('item_code'))
-		})
-
-	return frappe.db.sql(query, flt)
-
+	purchase_receipts = frappe.get_all('Purchase Receipt',
+		fields=fields,
+		filters=item_filters,
+		as_list=1
+	)
+	return purchase_receipts
```

Comment:
Where is the `item_code` condition ?

Reply:
added all the required condtions.

Reply:
Item code filter comes from queries.js where it filters at line 555